core/serialization/deserializer/montage-reviver.js

```diff
@@ -731,17 +728,13 @@ var MontageReviver = exports.MontageReviver = Montage.specialize(/** @lends Mont
                 bindingsToDeserializeDesc;
 
             if (bindingsToDeserialize) {
-                try {
-                    for (var i = 0, length = bindingsToDeserialize.length; i < length; i++) {
-                        bindingsToDeserializeDesc = bindingsToDeserialize[i];
-                        Bindings.deserializeObjectBindings(
-                            unitDeserializer.initWithContext(context),
-                            bindingsToDeserializeDesc.object,
-                            bindingsToDeserializeDesc.bindings
-                        );
-                    }
-                } catch (ex) {
-                    return Promise.reject(ex);
+                for (var i = 0, length = bindingsToDeserialize.length; i < length; i++) {
```

Comment:
how is that backward compatible with default being async?

Reply:
There is no async work being done here, we don't need to wrap the error in a promise.

Reply:
There is no async work being done here, we don't need to wrap the error in a promise.